core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -679,11 +769,70 @@ public int size() {
       return 0;
    }
 
+   @Override
+   public int size(IntSet segments) {
+      storesMutex.readLock().lock();
+      try {
+         checkStoreAvailability();
+         for (CacheLoader l : loaders) {
+            if (l instanceof SegmentedAdvancedLoadWriteStore)
+               return ((SegmentedAdvancedLoadWriteStore) l).size(segments);
+         }
+         for (CacheLoader l : loaders) {
+            if (l instanceof AdvancedCacheLoader)
+               return ((AdvancedCacheLoader) l).size();
```

Comment:
I'm not sure this fits with the description in the javadoc, "how many entries are in the store which map to the given segments" as we are returning the total number of entries for non-segmented stores. To accurately return the number of entries associated with the specified segments, we would obviously have to iterate over all the entries in a non-segmented store, which is probably overkill. Does it make more sense to make this method only return a value for segment aware stores?